core/core.go

```diff
@@ -120,8 +121,18 @@ func LimaVersionSupported() error {
 		return fmt.Errorf("error decoding 'limactl info' json: %w", err)
 	}
 	// remove pre-release hyphen
-	if str := strings.SplitN(values.Version, "-", 2); len(str) > 0 {
-		values.Version = str[0]
+	parts := strings.SplitN(values.Version, "-", 2)
+	if len(parts) > 0 {
+		values.Version = parts[0]
+	}
+
+	if parts[0] == "HEAD" {
+		version := parts[0]
+		if len(parts) > 1 {
+			version = parts[1]
+		}
+		logrus.Warnf("to avoid compatibility issues, ensure lima development version (%s) in use is not lower than %s", version, limaVersion)
+		return nil
```

Comment:
I just realised there is actually no need to extract any version details anymore.

```suggestion
	if parts[0] == "HEAD" {
		logrus.Warnf("to avoid compatibility issues, ensure lima development version (%s) in use is not lower than %s", values.Version, limaVersion)
		return nil
```